src/protocol/attribution/credit_capping.rs

```diff
@@ -23,6 +23,9 @@ use std::iter::{repeat, zip};
 /// ## Errors
 /// Fails if the multiplication protocol fails.
 ///
+/// ## Panics
```

Comment:
I'm not sure if we should panic *here*. This seems like something that we would ideally validate at the very beginning of the protocol, not delay until after we have done a bunch of work (like sorting).